PopPUNK/__main__.py

```diff
@@ -501,6 +520,34 @@ def main():
 
         nx.write_gpickle(genomeNetwork, args.output + "/" + os.path.basename(args.output) + '_graph.gpickle')
 
+    #******************************#
+    #*                            *#
+    #* within-strain analysis     *#
+    #*                            *#
+    #******************************#
+    sys.stderr.write("Checking options here\n\n")
+    if args.lineage_clustering:
+        sys.stderr.write("Mode: Identifying lineages within a clade\n\n")
+
+        # process rankings to use
+        rank_list = sorted([int(x) for x in args.R.split(',')])
+
+        # load distances
+        distances = None
```

Comment:
This is perl/C

Reply:
i.e. don't need to preassign as variable doesn't go out of scope in python

Reply:
Horrible witchcraft from a language that still can't work out things should be converted to strings within a print statement. Gone in effb8909e329c8ac57e1bc9aea09be1088f7021d.